core/drivers/crypto/stm32/cipher.c

```diff
@@ -362,6 +362,16 @@ static struct drvcrypt_cipher driver_cipher_saes = {
 
 TEE_Result stm32_register_cipher(enum stm32_cipher_ip_id cipher_ip)
 {
+	void *op = drvcrypt_get_ops(CRYPTO_CIPHER);
+
+	if (op == &driver_cipher_cryp) {
+		EMSG("Error CRYP peripheral already registered");
+		return TEE_ERROR_GENERIC;
+	} else if (op == &driver_cipher_saes) {
+		EMSG("Error SAES peripheral already registered");
+		return TEE_ERROR_GENERIC;
+	}
```

Comment:
I think this misses the case `op` has some non-NULL other value. It's not expected from now but could help in the long term.

Suggestion:
```c
	if (op) {
		EMSG("Error %s peripheral already registered for CRYPTO_CIPHER",
		     op == &driver_cipher_cryp ? "CRYP" :
		     op == &driver_cipher_saes ? "SAES" : "alternate");
		return TEE_ERROR_GENERIC;
	}
```